client_plugin/drivers/shared/dockerops/dockerops.go

```diff
@@ -0,0 +1,82 @@
+// Copyright 2017 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dockerops
+
+import (
+	"context"
+	log "github.com/Sirupsen/logrus"
+	dockerClient "github.com/docker/engine-api/client"
+	dockerTypes "github.com/docker/engine-api/types"
+)
+
+const (
+	// dockerAPIVersion: docker engine 1.24 and above support this api version
+	dockerAPIVersion = "v1.24"
+	// dockerUSocket: Unix socket on which Docker engine is listening
+	dockerUSocket = "unix:///var/run/docker.sock"
+)
+
+// DockerOps is the interface for docker host related operations
+type DockerOps struct {
+	Dockerd *dockerClient.Client
+}
+
+func NewDockerOps() *DockerOps {
+	var d *DockerOps
+
+	client, err := dockerClient.NewClient(dockerUSocket, dockerAPIVersion, nil, nil)
+	if err != nil {
+		log.WithFields(
+			log.Fields{"error": err},
+		).Error("Failed to create client for Docker ")
+		return nil
+	}
+
+	d = &DockerOps{
+		Dockerd: client,
+	}
+
+	return d
+}
+
+// VolumeCreate - create volume from docker host with specific volume driver
+func (d *DockerOps) VolumeCreate(volumeDriver string, volName string, options map[string]string) error {
```

Comment:
I don't agree since we are not just having volume operations here.
Below we have the function to launch the file sharing servers, which is not volume operation.
And later we will add more function like the ones to get swarm information here too.